src/org/labkey/test/pages/ReactAssayDesignerPage.java

```diff
@@ -0,0 +1,238 @@
+        return this;
+    }
+
+    public ReactAssayDesignerPage enableQCStates(boolean checked)
+    {
+        elementCache().qcEnabledCheckbox.set(checked);
+        return this;
+    }
+
+    public ReactAssayDesignerPage addTransformScript(File transformScript)
+    {
+        int index = Locator.xpath("//input[starts-with(@id, 'assay-design-protocolTransformScripts')]").findElements(getDriver()).size();
+        click(Locator.tagWithClass("span", "btn").containing("Add Script"));
+        return setTransformScript(transformScript, index);
+    }
+
+    public ReactAssayDesignerPage setTransformScript(File transformScript)
+    {
+        return setTransformScript(transformScript, 0);
+    }
+
+    public ReactAssayDesignerPage setTransformScript(File transformScript, int index)
+    {
+        assertTrue("Unable to locate the transform script: " + transformScript, transformScript.exists());
+        setFormElement(Locator.xpath("//input[@id='assay-design-protocolTransformScripts" + index + "']"), transformScript.getAbsolutePath());
+        return this;
+    }
+
+    public enum MetadataInputFormat implements OptionSelect.SelectOption
+    {
+        MANUAL,
+        FILE_BASED,
+        COMBINED;
+
+        @Override
+        public String getValue()
+        {
+            return name();
+        }
+
+        @Override
+        public String getText()
+        {
+            return null;
+        }
+    }
+
+    public DomainFormPanel goToFieldProperties(String title)
+    {
+        int attempts = 0; // don't try clicking next forever
+
+        DomainFormPanel panel = fieldProperties(title, true);
+        while (panel == null)
+        {
+            if (attempts > 10)
+                break;
+
+            clickNext();
+            panel = fieldProperties(title, true);
+            attempts++;
+        }
+
+        return panel;
+    }
+
+    public DomainFormPanel expandFieldProperties(String title)
+    {
+        DomainFormPanel panel = fieldProperties(title);
+        panel.expand();
+        return panel;
+    }
+
+    public ReactAssayDesignerPage clickBack()
+    {
+        scrollIntoView(elementCache().backBtn);
+        shortWait().until(ExpectedConditions.elementToBeClickable(elementCache().backBtn));
+        elementCache().backBtn.click();
+        return this;
+    }
+
+    public ReactAssayDesignerPage clickNext()
+    {
+        scrollIntoView(elementCache().nextBtn);
+        shortWait().until(ExpectedConditions.elementToBeClickable(elementCache().nextBtn));
+        elementCache().nextBtn.click();
+        return this;
+    }
+
+    public void clickFinish()
+    {
+        scrollIntoView(elementCache().finishButton);
+        shortWait().until(ExpectedConditions.elementToBeClickable(elementCache().finishButton));
+        String currentURL = getDriver().getCurrentUrl();
+        elementCache().finishButton.click();
+        afterSaveOrFinishClick(currentURL);
+    }
```

Comment:
I'm more convinced that these wizard workflow methods belong in a separate class. The pattern that is arising from this design is going to be painful to maintain; specifically, having to know the name of the last domain panel before clicking finish.